crates/catalog/src/mutator.rs

```diff
@@ -30,12 +30,52 @@ impl CatalogMutator {
         self.client.as_ref()
     }
 
+    /// Commit the catalog state.
+    /// This persists the state to the metastore.
+    pub async fn commit_state(
+        &self,
+        catalog_version: u64,
```

Comment:
does this expect the version to be the current version of the catalog upon which we want to apply the edits, or the version of the catalog we're trying update? I think the ambiguity/possibility for off-by-one errors in this case makes things confusing.

This is made further confusing because there's a version number in the catalog state, and I'm not sure how they should be compared, validated, or reconciled.

Reply:
yeah i found this a bit confusing too. I don't think this is directly related to this PR functionality though as this was already prevalent in the existing code. 

I do think we can clean up these interfaces a bit, but I think it'd detract from the core logic needed to fix this bug. Maybe we can open up an issue to follow up on this?

Reply:
> yeah i found this a bit confusing too. I don't think this is directly related to this PR functionality though as this was already prevalent in the existing code.

This is directly related to this pr. What should this version be when committing a catalog state? Does it match `state.version`?

Reply:
added some comments and cleaned this up to be clearer.